forensics/Copy-VmDigitalEvidence.ps1

```diff
@@ -42,152 +42,190 @@ $ErrorActionPreference = 'Stop'
+        
+        Write-Output "#################################"
+        Write-Output "Copy the Data Disk $dsnapshotName snapshot from source to blob container"
+        Write-Output "#################################"
+
+        $dsnapSasUrl = Grant-AzSnapshotAccess -ResourceGroupName $ResourceGroupName -SnapshotName $dsnapshotName -DurationInSecond 72000 -Access Read
+        $targetStorageContextBlob = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSABlob).Context
+        $targetStorageContextFile = (Get-AzStorageAccount -ResourceGroupName $destRGName -Name $destSAFile).Context
+
+        Write-Output "Start Copying Blob $dsnapshotName"
+        Start-AzStorageBlobCopy -AbsoluteUri $dsnapSasUrl.AccessSAS -DestContainer $destSAContainer -DestContext $targetStorageContextBlob -DestBlob "$dsnapshotName.vhd" -Force
+
+        Write-Output "Start Copying Fileshare"
+        Start-AzStorageFileCopy -AbsoluteUri $dsnapSasUrl.AccessSAS -DestShareName $destTempShare -DestContext $targetStorageContextFile -DestFilePath $dsnapshotName  -Force
+        
+        Write-Output "Waiting Fileshare Copy End"
+        Get-AzStorageFileCopyState -Context $targetStorageContextFile -ShareName $destTempShare -FilePath $dsnapshotName -WaitForComplete
+                
+        $ddiskpath = "$targetLinuxDir/$dsnapshotName"
+        Write-Output "Start Calculating HASH for $ddiskpath"
+        
+        $dhashfull = Invoke-Expression -Command "sha256sum $ddiskpath"
+        $dhash = $dhashfull.split(" ")[0]
+
+        Write-Output "Computed SHA-256: $dhash"
+
+        
+        
+        $BEKurl = $ddisk.EncryptionSettingsCollection.EncryptionSettings.DiskEncryptionKey.SecretUrl
+        Write-Output "#################################"
+        Write-Output "Disk Encryption Secret URL: $BEKurl"
+        Write-Output "#################################"
+        if ($BEKurl) {
+            $sourcekv = $BEKurl.Split("/")
+            $BEK = Get-AzKeyVaultSecret -VaultName  $sourcekv[2].split(".")[0] -Name $sourcekv[4] -Version $sourcekv[5]
+            Write-Output "Key value: $BEK"
+            Write-Output "Secret name: $dsnapshotName"
+            $BEK.Tags.Hash = $dhash
+            $secretName = $dsnapshotName.Replace("_","-")
+            Set-AzKeyVaultSecret -VaultName $destKV -Name $secretName -SecretValue $BEK.SecretValue -ContentType "BEK" -Tag $BEK.Tags
+        }
+        else {
+            Write-Output "Disk not encrypted"
+        }
+
+        Write-Output "#################################"
+        Write-Output "Data disk - Put hash value in Key Vault"
+        Write-Output "#################################"
+        $Secret = ConvertTo-SecureString -String $dhash -AsPlainText -Force
+        Set-AzKeyVaultSecret -VaultName $destKV -Name "$dsnapshotName-sha256" -SecretValue $Secret -ContentType "HASH"
+        Get-AzSubscription -SubscriptionId $destSubId | Set-AzContext

```

Comment:
We already do this (setting AzContext for the shared destination subscription id) right outside the loop.  Why would we want to do it again for each item in the loop?

Reply:
instance in the loop removed

Reply:
👍